### PR TITLE
[skipci] [Doctest] added contrib metrics ``AveragePrecision``, ``PrecisionRecallCurve``, ``ROC_AUC`` and ``RocCurve``

### PR DESCRIPTION
**Pull request by Ishan-Kumar2**
Addresses #2265

Description:
added doctests for contrib metrics ``AveragePrecision``, ``PrecisionRecallCurve``, ``ROC_AUC`` and ``RocCurve``
Not sure about the list of floats outputs, please let me know if there is a better way to handle those.
@ydcjeff @sdesrozis 

Check list:

- [X] New tests are added (if a new feature is added)
- [X] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -374,6 +374,11 @@ def train_step(engine, batch):
     ('fc', nn.Linear(2, 1))
 ]))
```
Comment (sdesrozis) [inline docs/source/conf.py:376]: This should be removed as we define the data in the doctest snippets.

File: ignite/contrib/metrics/average_precision.py [hunk 2]
```diff
@@ -33,14 +33,23 @@ class AveragePrecision(EpochMetric):
         AveragePrecision expects y to be comprised of 0's and 1's. y_pred must either be probability estimates or
         confidence values. To apply an activation to y_pred, use output_transform as shown below:
 
-        .. code-block:: python
+        .. testcode::
 
             def activated_output_transform(output):
```
Comment (sdesrozis) [inline ignite/contrib/metrics/average_precision.py:38]: This can be removed.

File: ignite/contrib/metrics/average_precision.py [hunk 3]
```diff
@@ -33,14 +33,23 @@ class AveragePrecision(EpochMetric):
...
                 y_pred, y = output
                 y_pred = torch.softmax(y_pred, dim=1)
                 return y_pred, y
+            y_pred = torch.Tensor([[3.89, 0.21], [0.30, 0.9], [0.46, 0.14], [0.16, 0.2]])
```
Comment (sdesrozis) [inline ignite/contrib/metrics/average_precision.py:42]: Use scores compliant with softmax

File: ignite/contrib/metrics/average_precision.py [hunk 4]
```diff
@@ -33,14 +33,23 @@ class AveragePrecision(EpochMetric):
...
+            y_true = torch.tensor([[1, 1], [1, 1], [1, 0], [0, 1]])
 
             avg_precision = AveragePrecision(activated_output_transform)
```
Comment (sdesrozis) [inline ignite/contrib/metrics/average_precision.py:45]:
```suggestion
            avg_precision = AveragePrecision()
```

File: ignite/contrib/metrics/precision_recall_curve.py [hunk 5]
```diff
@@ -35,14 +35,27 @@ class PrecisionRecallCurve(EpochMetric):
     PrecisionRecallCurve expects y to be comprised of 0's and 1's. y_pred must either be probability estimates
     or confidence values. To apply an activation to y_pred, use output_transform as shown below:
 
-    .. code-block:: python
+    .. testcode::
 
         def activated_output_transform(output):
```
Comment (sdesrozis) [inline ignite/contrib/metrics/precision_recall_curve.py:40]: This can be removed from the test, although it can be kept as a note.

File: ignite/contrib/metrics/precision_recall_curve.py [hunk 6]
```diff
@@ -35,14 +35,27 @@ class PrecisionRecallCurve(EpochMetric):
...
             y_pred, y = output
             y_pred = torch.sigmoid(y_pred)
             return y_pred, y
+        y_pred = torch.tensor([-3, 0.4, 0.9, 8])
```
Comment (sdesrozis) [inline ignite/contrib/metrics/precision_recall_curve.py:44]: Use compliant scores

File: ignite/contrib/metrics/precision_recall_curve.py [hunk 7]
```diff
@@ -35,14 +35,27 @@ class PrecisionRecallCurve(EpochMetric):
...
+        y_true = torch.tensor([0, 0, 1, 1])
+        prec_recall_curve = PrecisionRecallCurve(activated_output_transform)
```
Comment (sdesrozis) [inline ignite/contrib/metrics/precision_recall_curve.py:46]:
```suggestion
        prec_recall_curve = PrecisionRecallCurve()
```

File: ignite/contrib/metrics/roc_auc.py [hunk 8]
```diff
@@ -41,15 +41,18 @@ class ROC_AUC(EpochMetric):
...
     values. To apply an activation to y_pred, use output_transform as shown below:
 
-    .. code-block:: python
+    .. testcode::
 
-        def activated_output_transform(output):
-            y_pred, y = output
-            y_pred = torch.sigmoid(y_pred)
-            return y_pred, y
+        roc_auc = ROC_AUC(sigmoid_output_transform)
```
Comment (sdesrozis) [inline ignite/contrib/metrics/roc_auc.py:46]:
```suggestion
        roc_auc = ROC_AUC()
```

File: ignite/contrib/metrics/roc_auc.py [hunk 9]
```diff
@@ -88,15 +91,22 @@ class RocCurve(EpochMetric):
...
-    .. code-block:: python
+    .. testcode::
 
-        def activated_output_transform(output):
-            y_pred, y = output
-            y_pred = torch.sigmoid(y_pred)
-            return y_pred, y
+        roc_auc = RocCurve(sigmoid_output_transform)
+        roc_auc.attach(default_evaluator, 'roc_auc')
+        y_pred = torch.tensor([-3, 0.4, 0.9, 8])
```
Comment (sdesrozis) [inline ignite/contrib/metrics/roc_auc.py:98]: Use compliant scores

**Comment by sdesrozis** [comment]
@Ishan-Kumar2 Thanks a lot for your help ! I left some minor comments.

File: docs/source/conf.py (hunk 1 above)
Comment (Ishan-Kumar2) [reply docs/source/conf.py]: I think you are referring to the sigmoid function below these lines right? I have removed that since its not needed anymore.

File: ignite/contrib/metrics/average_precision.py [hunk 10]
```diff
@@ -33,14 +33,20 @@ class AveragePrecision(EpochMetric):
...
-            def activated_output_transform(output):
-                y_pred, y = output
-                y_pred = torch.softmax(y_pred, dim=1)
-                return y_pred, y
+            y_pred = torch.Tensor([[0.79, 0.21], [0.30, 0.70], [0.46, 0.54], [0.16, 0.84]])
+            y_true = torch.tensor([[1, 1], [1, 1], [0, 1], [0, 1]])
 
-            avg_precision = AveragePrecision(activated_output_transform)
+            avg_precision = AveragePrecision()
+            #The ``output_transform`` arg of the metric can be used to perform a softmax on the ``y_pred``.
```
Comment (sdesrozis) [inline ignite/contrib/metrics/average_precision.py:42]: Thinking about it, it is worth to keep the current example to show the way to respect the data format 

```python 
AveragePrecision expects y to be comprised of 0's and 1's. y_pred must either be probability estimates or
        confidence values. To apply an activation to y_pred, use output_transform as shown below:

        .. code-block:: python

            def activated_output_transform(output):
                y_pred, y = output
                y_pred = torch.softmax(y_pred, dim=1)
                return y_pred, y

            avg_precision = AveragePrecision(activated_output_transform)
```

**Review by sdesrozis — APPROVED** [review]
@Ishan-Kumar2 Thank you ! LGTM
